### ISSUE TITLE
the code of git branch of SDXL has a bug about metadata：type conversion error caused by misaligned parameters

### ISSUE THREAD

**Issue by lilyzlt** [kohya-ss/sd-scripts#734]
![image](https://github.com/kohya-ss/sd-scripts/assets/15058321/db150fcd-745a-45db-a3d7-ad369fb77451)
![image](https://github.com/kohya-ss/sd-scripts/assets/15058321/b060131d-836b-4836-aa45-860a33fe6482)
as the screenshot,  code train_util.py function of "sai_model_spec.build_metadata" , because the position missing of "merged_from", then cause the type error, 
![企业微信截图_f0d2911b-8035-4b41-9fb1-152735f3e3c6](https://github.com/kohya-ss/sd-scripts/assets/15058321/aac31462-7b7b-4683-a2e5-c0a78b834a24)
the error"argument 'metadata': 'int' object cannot be converted to 'PyString'"

update this part code to as below , then fix the error 
`    metadata = sai_model_spec.build_metadata(`   
      `state_dict,`
       ` v2,`
        `v_parameterization,`
        `sdxl,`
        `lora,`
        `textual_inversion,`
        `timestamp,`
        `title,`
       `reso,`
        `is_stable_diffusion_ckpt,`
        `args.metadata_author,`
        `args.metadata_description,`
        `args.metadata_license,`
        `args.metadata_tags,`
        `None,`
       `timesteps,`
        `args.clip_skip,  # None or int`
    `)`

**Comment by lilyzlt** [comment]
it still the same error after I update "pip install -e .", 
<img width="653" alt="截屏2023-08-08 22 40 53" src="https://github.com/kohya-ss/sd-scripts/assets/15058321/f6787346-33a1-40dc-b410-fef501ae0c56">
<img width="1438" alt="image" src="https://github.com/kohya-ss/sd-scripts/assets/15058321/cfdeee28-45fc-465d-96a0-fa070e39f822">
first of all, this happend at sdxl branch code,
this error is because of that string arg 'merged_from' is missing, and the next arg is int type.

**Comment by lilyzlt** [comment]
> Thank you for reporting. The line #713 in current `train_network.py` is different from your screen shot.
> 
> https://github.com/kohya-ss/sd-scripts/blob/4e339bb101e76162a78f12d645483595ef2fe7d8/train_network.py#L713
> 
> I think you might be using an older version. Please update the repo to make sure.

yeah, I know about that, I add 2 "print()"code  to find out what case the error of `the error"argument 'metadata': 'int' object cannot be converted to 'PyString'"`,after I deleted that print, the code is the same.
<img width="873" alt="image" src="https://github.com/kohya-ss/sd-scripts/assets/15058321/a5e32361-b922-47ab-8d97-cc20de804acd">

**Comment by lilyzlt** [comment]
> I understand about the line number.
> 
> As you know, keyword arguments should be order-independent. Perhaps there is a bug elsewhere. Could you please share the command line arguments and the .toml file?
the Command as below, 
`/data/miniconda3/bin/python', 'sd-scripts/sdxl_train_network.py', 
'--pretrained_model_name_or_path=/data/stable-diffusion-xl-base-1.0/sd_xl_base_1.0.safetensors', '--train_data_dir=/data/Images/dog','--output_dir=/daat/Lora/', '--logging_dir=./logs', '--resolution=1024,1024', '--max_train_epochs=10', '--learning_rate=1e-4', '--lr_scheduler=constant_with_warmup', '--lr_warmup_steps=10', '--lr_scheduler_num_cycles=1', '--output_name=sdxl_demo', '--train_batch_size=8', '--save_every_n_epochs=1', 
'--save_precision=bf16', '--seed=1337', '--max_token_length=225', '--caption_extension=.txt', '--save_model_as=safetensors', '--keep_tokens=3', '--xformers', '--no_half_vae', 
'--cache_text_encoder_outputs', '--cache_text_encoder_outputs_to_disk', '--bucket_reso_steps=32', '--network_module=networks.lora', '--network_dim=32', '--network_alpha=1', '--unet_lr=1e-5',
'--text_encoder_lr=2e-4', '--training_comment=sdxl_training_from_venus', '--save_state', '--network_train_unet_only', '--persistent_data_loader_workers', '--cache_latents', 
'--cache_latents_to_disk', '--network_dropout=0.1', '--scale_weight_norms=1.0', '--network_args', 'rank_dropout=0', 'module_dropout=0', '--network_args', 'conv_dim=8', 'conv_alpha=1', 
'algo=loha', 'dropout=0', '--optimizer_type=AdamW8bit', '--optimizer_args', 'weight_decay=0.01', '--noise_offset=0.0375', '--adaptive_noise_scale=0.0375', '--vae_batch_size=4', 
'--gradient_checkpointing', '--clip_skip=2'`

while I didn't use  the .toml file, the data looks like this, may it be related with that, no I mean .toml file cause the error?
![image](https://github.com/kohya-ss/sd-scripts/assets/15058321/39aa17b1-2656-4e47-a617-ab47dd3d814d)

**Comment by lilyzlt** [comment]
> 

I'm sorry I just saw the commit "fix crashing in saving lora with clipskip" that fix the problem, I tried~ thx 
https://github.com/kohya-ss/sd-scripts/commit/6f80fe17fcac026ab85004d5a701835c14f5da84
